backend/api/notifications.py

```diff
@@ -0,0 +1,92 @@
+from flask import request, jsonify, make_response, g, Blueprint
+from flask_security import login_required, current_user
+import time
+import pymongo
+import uuid
+
+
+notifications = Blueprint('api_notifications', __name__)
+
+
+def add_notification(creator, userlist, project_id, operation, reason='', saved_search_id=''):
+    date = time.strftime("%Y-%m-%d %H:%M:%S")
+    for user in userlist:
+        if user not in (creator):
+            g.notifications.insert_one({
+                '_id': str(uuid.uuid4()),
+                'creator': creator,
+                'user_id': user,
```

Comment:
Feel free to make them as Embedded documents... Using pymongo only for convenience